src/functions.php

```diff
@@ -293,3 +291,26 @@ function _checkTypehint(callable $callback, \Throwable $reason): bool
     // If we look for a full match (intersection) and did not return early, we matched all types and are true
     return $isTypeUnion ? false : true;
 }
+
+/**
+ * @internal
+ */
+function matchesType($reason, $type): bool
+{
+    return ($type->isBuiltin() && \gettype($reason) === $type->getName())
+        || (new \ReflectionClass($type->getName()))->isInstance($reason);
+}
+
+/**
+ * @internal
+ */
+function matchesAllTypes($reason, $types): bool
```

Comment:
This function and the one above do not have type declarations. See note above, perhaps we don't need these functions in the first place?